vault/resource_approle_auth_backend_role_secret_id.go

```diff
@@ -225,19 +223,21 @@ func approleAuthBackendRoleSecretIDRead(d *schema.ResourceData, meta interface{}
 	}
 
 	var cidrs []string
-	switch resp.Data["cidr_list"].(type) {
-	case string:
-		if resp.Data["cidr_list"].(string) != "" {
-			cidrs = strings.Split(resp.Data["cidr_list"].(string), ",")
-		}
-	case []interface{}:
-		v := resp.Data["cidr_list"].([]interface{})
-		cidrs = make([]string, 0, len(v))
-		for _, i := range v {
-			cidrs = append(cidrs, i.(string))
+	if cidrList := resp.Data["cidr_list"]; cidrList != nil {
+		switch cidrList.(type) {
+		case string:
+			if cidrList.(string) != "" {
```

Comment:
These casts can be avoided with a switch assignment: https://play.golang.org/p/V8rQXqvNG3h

Reply:
Nice! Updated in https://github.com/hashicorp/terraform-provider-vault/pull/1230/commits/73bc5acbae76d28cc0a2f849f5accfc2d36c5a96